sdk/Sdk.Generators/FunctionExecutor/FunctionExecutorGenerator.Emitter.cs

```diff
@@ -143,79 +143,86 @@ private static string GetMethodBody(IEnumerable<ExecutableFunction> functions, b
             {
                 var sb = new StringBuilder();
                 sb.Append(
-                   $$"""
+                   $"""
                 var inputBindingFeature = context.Features.Get<global::Microsoft.Azure.Functions.Worker.Context.Features.IFunctionInputBindingFeature>();
                             var inputBindingResult = await inputBindingFeature.BindFunctionInputAsync(context);
                             var inputArguments = inputBindingResult.Values;
-                {{(anyDefaultExecutor ? $"            _defaultExecutor = new Lazy<global::Microsoft.Azure.Functions.Worker.Invocation.IFunctionExecutor>(() => CreateDefaultExecutorInstance(context));{Environment.NewLine}" : string.Empty)}}
+                {(anyDefaultExecutor ? $"            _defaultExecutor = new Lazy<global::Microsoft.Azure.Functions.Worker.Invocation.IFunctionExecutor>(() => CreateDefaultExecutorInstance(context));{Environment.NewLine}" : string.Empty)}
                 """);
 
-                bool first = true;
-
                 foreach (ExecutableFunction function in functions)
                 {
                     var fast = function.Visibility == FunctionMethodVisibility.Public;
                     sb.Append($$"""
 
-                        {{(first ? string.Empty : "else ")}}if (string.Equals(context.FunctionDefinition.EntryPoint, "{{function.EntryPoint}}", StringComparison.Ordinal))
+                        if (string.Equals(context.FunctionDefinition.EntryPoint, "{{function.EntryPoint}}", StringComparison.Ordinal))
                         {
                            {{(fast ? EmitFastPath(function) : EmitSlowPath())}}
+                            goto end;
                         }
+
             """);
-                    first = false;
                 }
 
+                sb.Append(
+            """
+
+                        end:
+                        return;
+            """);
                 return sb.ToString();
             }
 
             private static string EmitFastPath(ExecutableFunction function)
             {
                 var sb = new StringBuilder();
-                int functionParamCounter = 0;
```

Comment:
Made changes to this method to improve readability and move declarations closer to use.